doc/loganalyzer_to_pytest.md

```diff
@@ -0,0 +1,263 @@
+
+```LogAnayzer class object``` - Directly imported loganalyzer class with similar interface that is provided by the "Autouse Pytest fixture". Can be used in pytest hooks.  Proposal is to avoid direct usage.
+
+#### How to disable loganalyzer for all run or for specific test case
+- For all tests run use ```--disable_loganalyzer``` option in the console.
+- For specific test case use @disable_loganalyzer decorator
+
+#### disable_loganalyzer decorator
+Its role is to somehove indicate loganalyzer fixture to skip log analysis for specific test case. Proposed simple approach to create attribute in the pytest module and use it as global variable beatween all test modules.
+
+#### Example how to skip specific test case
+tests/conftest.py
+```
+import pytest
+from ansible_host import ansible_host
+
+pytest.disable_log_analyzer = False # Log analyzer will be turned on for all test cases
+
+@pytest.fixture(autouse=True)
+def loganalyzer(ansible_adhoc, testbed):
+    loganalyzer = LogAnalyzer(ansible_host=ansible_host(ansible_adhoc, testbed['dut']), marker_prefix="loganalyzer")
+    if not pytest.disable_loganalyzer:
+	    marker = loganalyzer.init()
+	yield loganalyzer
+	if not pytest.disable_loganalyzer:
+	    result = loganalyzer.analyze(marker)
+		if not result:
+			pytest.fail("Log analyzer failed.")
+		assert result["total"]["match"] == 0, "Found errors: {}".format(result)
+```
+
+tests/lib/helpers.py
+```
+import pytest
+def disable_loganalyzer(func):
+    def wrapper(*args, **kwargs):
+        pytest.disable_loganalyzer = True
+        try:
+            func(*args, **kwargs)
+        finally:
+            pytest.disable_loganalyzer = False
+    return wrapper
+```
+
+Any test case which requires disabled loganalyzer should use ```disable_loganalyzer``` decorator:
+```
+@disable_loganalyzer
+def test(x, y):
+    assert x == y
+```
+
+#### Development
+Module "loganalyzer.py" with class "Loganalyzer".
+
+"Loganalyzer" class interface:
+- **__init__**(ansible_host: ansible_host, marker_prefix, dut_run_dir="/tmp")
+- **load_common_config()** - Clear previous configured match, expect and ignore. Load regular expressions from common configuration files: match, expect and ignore which are located in some configuration directory or with current module. Save loaded configuration to the self.match_regex, self.expect_regex and self.ignore_regex attributes.
+- **parse_regexp_file(file_path)** - Read and parse regular expressions from specified file. Return list of strings of defined regular expressions.
+- **run_cmd(callback, *args, **kwargs)** - Call function and analyze DUT syslog during function execution. Return the same result as "analyze" function.
+- **init()** - Add start marker to the DUT syslog. Generated marker format: marker_prefix + "%Y-%m-%d-%H:%M:%S". Return generated prefix.

```

Comment:
In the current design, the marker includes testcase name as well. Are we not using it here?

Reply:
During loganalyzer object instantiation, it receives "marker_prefix" parameter, like in example:
LogAnalyzer(ansible_host=ans_host, marker_prefix="test_loganalyzer")
This can be test case name.
During test cases run, autouse fixture can change this per each test case by calling "update_marker_prefix" method, like in example:
log.update_marker_prefix("test_fdb").
So there is possibility to pass test case name for loganalyzer marker.